### DISCUSSION TITLE
Theming: `sleep(float fSeconds)` doesn't.

### DISCUSSION THREAD

**Discussion by ValVoice** [stepmania/stepmania#2136] [Q&A]
I'm continually checking for a value to change because I'm too dumb to come up with an alternative method. However, the method I'm using to wait before the next check *doesn't* wait until the next check.

This is the code I'm using.

    WaitForCheckCommand=function(self)
        attempts = attempts + 1;
        Trace(string.format("Checked %2d times. Aux is still AUX_DEFAULT", attempts));
        self:sleep(1.0);
        self:playcommand("CheckAux");
    end;

This should make the Actor pause 1 second before firing the `CheckAuxCommand` again, or so I thought. But looking at the logs, it seems like it's not waiting at all.

How do I silve this problem by either making the Actor actually wait, or by using a better method to get the index of a `Scroll` inside the `ScreenTitleMenu` scroller?

**Comment by JoseVarelaP** [comment]
Using `playcommand` on a command will play instantly, which will ignore all times from the tween. You should consider using `queuecommand` instead, which will wait for whatever tween time happens before it reaches the command.

```lua
WaitForCheckCommand=function(self)
    attempts = attempts + 1;
    Trace(string.format("Checked %2d times. Aux is still AUX_DEFAULT", attempts));
    self:sleep(1.0);
    self:queuecommand("CheckAux");
end;
```
